### ISSUE TITLE
vim.fault.FileNotFound for a volume while creating the said volume

### ISSUE THREAD

**Issue by anchal-agrawal** [vmware/vic#5301]
Seen in https://ci.vcna.io/vmware/vic/11044 in the `Create with named volume` test in `1-04-Docker-Create`:
```
Error response from daemon: Server error from portlayer: [POST /storage/volumes][500] createVolumeInternalServerError  &{Code:500 Message:ServerFaultCode: File [datastore1] test/volumes/test-named-vol was not found}
```

Personality:
```
May 30 2017 23:41:24.058Z DEBUG [ END ] [github.com/vmware/vic/lib/apiservers/engine/backends.(*Volume).volumeCreate:227] [13.027466ms] 
May 30 2017 23:41:24.058Z DEBUG [ END ] [github.com/vmware/vic/lib/apiservers/engine/backends.(*ContainerProxy).AddVolumesToContainer:359] [13.08983ms] 6caac31cec1a6c4b4f17d5bb97a2659f
May 30 2017 23:41:24.058Z DEBUG [ END ] [github.com/vmware/vic/lib/apiservers/engine/backends.(*Container).containerCreate:619] [56.73178ms] Container.containerCreate
May 30 2017 23:41:24.058Z DEBUG [ END ] [github.com/vmware/vic/lib/apiservers/engine/backends.(*Container).ContainerCreate:557] [56.975609ms] 
May 30 2017 23:41:24.058Z ERROR Handler for POST /v1.25/containers/create returned error: Server error from portlayer: [POST /storage/volumes][500] createVolumeInternalServerError  &{Code:500 Message:ServerFaultCode: File [datastore1] test/volumes/test-named-vol was not found}
```

Portlayer:
```
May 30 2017 23:41:24.046Z DEBUG [BEGIN] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*StorageHandlersImpl).CreateVolume:330] storage_handlers.CreateVolume
May 30 2017 23:41:24.046Z DEBUG op=251.21 (delta:2.695µs): [NewOperation] op=251.21 (delta:885ns) [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*StorageHandlersImpl).CreateVolume:354]
May 30 2017 23:41:24.046Z INFO  Creating directory [datastore1] test/volumes/test-named-vol
May 30 2017 23:41:24.056Z DEBUG Creating [datastore1] test/volumes/test-named-vol error: ServerFaultCode: File [datastore1] test/volumes/test-named-vol was not found
May 30 2017 23:41:24.056Z ERROR storagehandler: VolumeCreate error: soap.soapFaultError{fault:(*soap.Fault)(0xc42071a050)}
May 30 2017 23:41:24.056Z DEBUG [ END ] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*StorageHandlersImpl).CreateVolume:330] [10.491112ms] storage_handlers.CreateVolume
```

hostd:
```
2017-05-30T23:41:24.972Z info hostd[7E8D6B70] [Originator@6876 sub=Vimsvc.TaskManager opID=1a7b5837 user=droneci] Task Created : haTask--vim.FileManager.makeDirectory-1956356
2017-05-30T23:41:24.975Z info hostd[7E8D6B70] [Originator@6876 sub=Nfcsvc opID=1a7b5837 user=droneci] Create requested for /vmfs/volumes/58af7eb5-d9fc5174-6127-0cc47a9be492/test/volumes/test-named-vol
2017-05-30T23:41:24.976Z info hostd[7E8D6B70] [Originator@6876 sub=Default opID=1a7b5837 user=droneci] AdapterServer caught exception: vim.fault.FileNotFound
2017-05-30T23:41:24.976Z info hostd[7E8D6B70] [Originator@6876 sub=Vimsvc.TaskManager opID=1a7b5837 user=droneci] Task Completed : haTask--vim.FileManager.makeDirectory-1956356 Status error
2017-05-30T23:41:24.976Z info hostd[7E8D6B70] [Originator@6876 sub=Solo.Vmomi opID=1a7b5837 user=droneci] Activation [N5Vmomi10ActivationE:0x7b1c6468] : Invoke done [makeDirectory] on [vim.FileManager:ha-nfc-file-manager]
2017-05-30T23:41:24.976Z verbose hostd[7E8D6B70] [Originator@6876 sub=Solo.Vmomi opID=1a7b5837 user=droneci] Arg name:
--> "[datastore1] test/volumes/test-named-vol"
2017-05-30T23:41:24.976Z verbose hostd[7E8D6B70] [Originator@6876 sub=Solo.Vmomi opID=1a7b5837 user=droneci] Arg datacenter:
--> 'vim.Datacenter:ha-datacenter'
2017-05-30T23:41:24.976Z verbose hostd[7E8D6B70] [Originator@6876 sub=Solo.Vmomi opID=1a7b5837 user=droneci] Arg createParentDirectories:
--> false
2017-05-30T23:41:24.976Z info hostd[7E8D6B70] [Originator@6876 sub=Solo.Vmomi opID=1a7b5837 user=droneci] Throw vim.fault.FileNotFound
2017-05-30T23:41:24.976Z info hostd[7E8D6B70] [Originator@6876 sub=Solo.Vmomi opID=1a7b5837 user=droneci] Result:
--> (vim.fault.FileNotFound) {
-->    faultCause = (vmodl.MethodFault) null,
-->    file = "[datastore1] test/volumes/test-named-vol",
-->    msg = ""
--> }
```

**This looks similar to https://github.com/vmware/vic/issues/4297 except I didn't find hostd logs indicating that another vic-machine operation attempted to delete this volume/store concurrently. Also, in that build, two successive (valid) volume create tests failed, but in this case just this volume was unable to be created and subsequent volume create ops succeeded.**

Log bundle:
[Test-Cases.Group1-Docker-Commands.1-04-Docker-Create-VCH-11044-6681-container-logs.zip](https://github.com/vmware/vic/files/1040230/Test-Cases.Group1-Docker-Commands.1-04-Docker-Create-VCH-11044-6681-container-logs.zip)

**Comment by anchal-agrawal** [comment]
Estimated as a 3 for the investigative aspect.

**Comment by chengwang86** [comment]
Also seen in test 3-02-voting-app here https://ci.vcna.io/vmware/vic/11246

Persona log:
```
Jun  5 2017 21:27:36.878Z DEBUG [BEGIN] [github.com/vmware/vic/lib/apiservers/engine/backends.(*Volume).volumeCreate:227]
Jun  5 2017 21:27:36.878Z INFO  Finalized model for volume create request to portlayer: &models.VolumeRequest{Capacity:-1, Driver:"vsphere", DriverArgs:map[string]string{"flags":"rw", "container":"redis", "image":"redis:alpine"}, Metadata:map[string]string{"DockerMetaData":"{\"Driver\":\"vsphere\",\"DriverOpts\":{\"container\":\"redis\",\"flags\":\"rw\",\"image\":\"redis:alpine\"},\"Name\":\"cbaaddc3-4a35-11e7-9dd0-000c297828ef\",\"Labels\":null,\"AttachHistory\":[\"redis\"],\"Image\":\"redis:alpine\"}"}, Name:"cbaaddc3-4a35-11e7-9dd0-000c297828ef", Store:"default"}
Jun  5 2017 21:27:36.937Z DEBUG [ END ] [github.com/vmware/vic/lib/apiservers/engine/backends.(*Volume).volumeCreate:227] [59.196079ms] 
Jun  5 2017 21:27:36.937Z DEBUG [ END ] [github.com/vmware/vic/lib/apiservers/engine/backends.(*ContainerProxy).AddVolumesToContainer:359] [59.259797ms] afe5f033616fdbd40ce5b2c8e85df84a
Jun  5 2017 21:27:36.937Z DEBUG [ END ] [github.com/vmware/vic/lib/apiservers/engine/backends.(*Container).containerCreate:619] [360.064312ms] Container.containerCreate
Jun  5 2017 21:27:36.937Z DEBUG [ END ] [github.com/vmware/vic/lib/apiservers/engine/backends.(*Container).ContainerCreate:557] [360.207392ms] 
Jun  5 2017 21:27:36.937Z ERROR Handler for POST /v1.25/containers/create returned error: Server error from portlayer: [POST /storage/volumes][500] createVolumeInternalServerError  &{Code:500 Message:ServerFaultCode: File [datastore1] test/volumes/cbaaddc3-4a35-11e7-9dd0-000c297828ef was not found}
```

PL log
```
Jun  5 2017 21:27:36.878Z DEBUG op=283.103 (delta:3.175µs): [NewOperation] op=283.103 (delta:1.336µs) [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*StorageHandlersImpl).CreateVolume:354]
Jun  5 2017 21:27:36.878Z INFO  Creating directory [datastore1] test/volumes/cbaaddc3-4a35-11e7-9dd0-000c297828ef
Jun  5 2017 21:27:36.918Z DEBUG vSphere Event Creating result-4175f89d7103 on host ns539601.ip-144-217-72.net in ha-datacenter for eventID(308309) ignored by the event collector
Jun  5 2017 21:27:36.918Z DEBUG vSphere Event Creating worker-94df235c186f on host ns539601.ip-144-217-72.net in ha-datacenter for eventID(308310) ignored by the event collector
Jun  5 2017 21:27:36.937Z DEBUG Creating [datastore1] test/volumes/cbaaddc3-4a35-11e7-9dd0-000c297828ef error: ServerFaultCode: File [datastore1] test/volumes/cbaaddc3-4a35-11e7-9dd0-000c297828ef was not found
Jun  5 2017 21:27:36.937Z ERROR storagehandler: VolumeCreate error: soap.soapFaultError{fault:(*soap.Fault)(0xc42085d7c0)}
Jun  5 2017 21:27:36.937Z DEBUG [ END ] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*StorageHandlersImpl).CreateVolume:330] [58.841193ms] storage_handlers.CreateVolume
```

hostd.log
```
2017-06-05T21:27:37.430Z info hostd[2C3CDB70] [Originator@6876 sub=Hostsvc.DatastoreSystem] RefreshVdiskDatastores: Done refreshing datastores.
2017-06-05T21:27:37.430Z info hostd[2F180B70] [Originator@6876 sub=Nfcsvc opID=93aa8a63 user=droneci] Create requested for /vmfs/volumes/589f9716-34cbc097-9753-0cc47a9adeac/test/volumes/cbaaddc3-4a35-11e7-9dd0-000c297828ef
2017-06-05T21:27:37.437Z info hostd[2F180B70] [Originator@6876 sub=Default opID=93aa8a63 user=droneci] AdapterServer caught exception: vim.fault.FileNotFound
2017-06-05T21:27:37.437Z info hostd[2F180B70] [Originator@6876 sub=Vimsvc.TaskManager opID=93aa8a63 user=droneci] Task Completed : haTask--vim.FileManager.makeDirectory-6208666 Status error
2017-06-05T21:27:37.437Z info hostd[2F180B70] [Originator@6876 sub=Solo.Vmomi opID=93aa8a63 user=droneci] Activation [N5Vmomi10ActivationE:0x319e62a8] : Invoke done [makeDirectory] on [vim.FileManager:ha-nfc-file-manager]
2017-06-05T21:27:37.437Z verbose hostd[2F180B70] [Originator@6876 sub=Solo.Vmomi opID=93aa8a63 user=droneci] Arg name:
--> "[datastore1] test/volumes/cbaaddc3-4a35-11e7-9dd0-000c297828ef"
2017-06-05T21:27:37.437Z verbose hostd[2F180B70] [Originator@6876 sub=Solo.Vmomi opID=93aa8a63 user=droneci] Arg datacenter:
--> 'vim.Datacenter:ha-datacenter'
2017-06-05T21:27:37.437Z verbose hostd[2F180B70] [Originator@6876 sub=Solo.Vmomi opID=93aa8a63 user=droneci] Arg createParentDirectories:
--> false
2017-06-05T21:27:37.437Z info hostd[2F180B70] [Originator@6876 sub=Solo.Vmomi opID=93aa8a63 user=droneci] Throw vim.fault.FileNotFound
2017-06-05T21:27:37.437Z info hostd[2F180B70] [Originator@6876 sub=Solo.Vmomi opID=93aa8a63 user=droneci] Result:
--> (vim.fault.FileNotFound) {
-->    faultCause = (vmodl.MethodFault) null,
-->    file = "[datastore1] test/volumes/cbaaddc3-4a35-11e7-9dd0-000c297828ef",
-->    msg = ""
--> }
```

**Comment by emlin** [comment]
The problem should be same to #4297 and the fix for #4297 does not update this test case to use unique volume name.

**Comment by emlin** [comment]
@anchal-agrawal Still found following message in hostd log:
`2017-05-30T23:41:24.608Z info hostd[7E751B70] [Originator@6876 sub=Nfcsvc opID=1a7b57fb user=droneci] file delete force (/vmfs/volumes/58af7eb5-d9fc5174-6127-0cc47a9be492/test/volumes -> /vmfs/volumes/58af7eb5-d9fc5174-6127-0cc47a9be492/test/volumes) by 'droneci'`

**Comment by emlin** [comment]
okay, actually the fix for #4297 already changed the volume store name with VCH name, so not possible to have same volume store path same to this issue in concurrent test env.

@anchal-agrawal I think this issue is already fixed with the above reason, and we didn't see it happen after Jun 13, that is the fix for #4297 is merged. So suggest to close this issue as already fixed.
